### PR TITLE
Use fnv hasher for HashMap and HashSet.

### PR DESCRIPTION
**Pull request by Object905**
With ecs, keys for hashmaps and sets are usually some kind of id, hence is very small. So fnv hash function will give significant performance in this case. [Here is compassion](http://cglab.ca/~abeinges/blah/hash-rs/). Also default hasher resistant to HashDoS attacks, but we don't need it anyway.

### CODE REVIEW

File: src/ecs/resources/input.rs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 //! World resource that handles all user input.
 
-use std::collections::hash_map::{Entry, HashMap, Keys};
+use std::collections::hash_map::{Entry, Keys};
+use fnv::FnvHashMap;
```
Comment (ebkalderon) [inline src/ecs/resources/input.rs]: Please try to keep the `use` statements in alphabetical order.

File: src/asset_manager/asset_manager.rs [hunk 2]
```diff
@@ -6,7 +6,7 @@ use dds::DDS;
 use gfx::texture::{AaMode, Kind};
 use imagefmt::{ColFmt, Image, read_from};
 use std::any::{Any, TypeId};
-use std::collections::HashMap;
+use fnv::FnvHashMap;
```
Comment (ebkalderon) [inline src/asset_manager/asset_manager.rs]: Please try to keep the `use` statements in alphabetical order.

File: Cargo.toml [hunk 3]
```diff
@@ -36,6 +36,7 @@ glutin = "0.7"
 imagefmt = "4.0"
 specs = "0.7"
 wavefront_obj = "5.0"
+fnv = "1.0.5"
```
Comment (ebkalderon) [inline Cargo.toml]: Please try to keep the dependencies in alphabetical order. Also, I would recommend omitting the `.5` at the end so Cargo can automatically update patch releases.

File: src/lib.rs [hunk 4]
```diff
@@ -77,6 +77,7 @@ extern crate imagefmt;
 extern crate num_cpus;
 extern crate specs;
 extern crate wavefront_obj;
+extern crate fnv;
```
Comment (ebkalderon) [inline src/lib.rs]: Please try to keep the `extern crate` statements in alphabetical order.

File: src/renderer/Cargo.toml [hunk 5]
```diff
@@ -25,6 +25,7 @@ gfx_window_glutin = "0.14"
 glutin = "0.7"
 mopa = "0.2"
 specs = "0.7"
+fnv = "1.0.5"
```
Comment (ebkalderon) [inline src/renderer/Cargo.toml]: Please try to keep the dependencies in alphabetical order. Also, I would recommend omitting the `.5` at the end so Cargo can automatically update patch releases.

File: AUTHORS.md [hunk 6]
```diff
@@ -16,3 +16,4 @@
 * Robbie Cooper
 * Scott Corbeil
 * White-Oak
+* Konstantin Zverev
```
Comment (ebkalderon) [inline AUTHORS.md]: Please try to keep the authors list in alphabetical order.

File: src/renderer/src/lib.rs [hunk 7]
```diff
@@ -17,7 +18,7 @@ pub mod target;
 
 use specs::{Component, VecStorage};
 use std::any::TypeId;
-use std::collections::HashMap;
+use fnv::FnvHashMap;
```
Comment (ebkalderon) [inline src/renderer/src/lib.rs]: Please try to keep the `use` statements in alphabetical order.

**Comment by Aceeri** [comment]
Are we sure we don't want to keep the hashmap/sets described as the `FnvHashMap/Set` types? It might be a little confusing to modify in the future since you'd normally guess the `HashMap` type is the std one.

**Comment by vitvakatu** [comment]
@Aceeri but you can always take a look at the first lines of file and find the truth.

**Comment by Aceeri** [comment]
Yeah but it feels like an unnecessary abstraction and is harder to understand by reading only those portions of the code. Do we gain anything from abstracting it like this?

**Comment by vitvakatu** [comment]
Well, it makes sense. I agree with you

**Comment by ebkalderon** [comment]
@Aceeri Remember that the `FnvHashMap` and `FnvHashSet` types are simply type aliases to the `std::collections::HashMap` and `HashSet` types. Using `as HashMap` is perfectly fine in this case.

**Comment by Aceeri** [comment]
Ah I see, I was thinking it was another implementation of a hashmap/set.

**Pull request merged**